### PR TITLE
Added information related to managing secrets in values files

### PR DESCRIPTION
**Pull request by hbisht-RH-ccs**
Added information related to managing secrets in values files

### CODE REVIEW

**Comment by mbaldessari** [comment]
This is an automated message:

You can preview this docs PR at http://435.docs-pr.validatedpatterns.io
Note that they get generated every five minutes, so please wait a bit.

File: content/learn/secrets-management-in-the-Validated-Patterns-framework.adoc [hunk 1]
```diff
@@ -0,0 +1,33 @@
+---
+menu:
+  learn:
+    parent: Values Files
```
Comment (mhjacks) [inline content/learn/secrets-management-in-the-Validated-Patterns-framework.adoc:4]: Should the parent here rather be secrets?

**Review by mhjacks — CHANGES_REQUESTED** [review]
I think this should go under the Secrets section rather than value files. Willing to discuss further, though.

File: content/learn/secrets-management-in-the-Validated-Patterns-framework.adoc (hunk 1 above)
Comment (hbisht-RH-ccs) [reply content/learn/secrets-management-in-the-Validated-Patterns-framework.adoc:4]: Based on Jira [Telcodocs-1597](https://issues.redhat.com/browse/TELCODOCS-1597), I've added the information under the values files section, as it is part of the documentation around values and charts. However, I understand that we're providing detailed information about managing secrets in validating patterns and that there is a separate section dedicated to secrets.
@danmacpherson, could you please confirm if this information should stay under the values section or be moved to the secrets section?

File: modules/basic-secrets-configuration-in-a-clustergroup.adoc [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+:imagesdir: ../../images
+
+[id="basic-secrets-configuration-in-a-clustergroup"]
+= Basic secrets configuration in a ClusterGroup
+
+The secrets required for a pattern are not directly represented in the ClusterGroup configuration. Instead, the `values-secret.yaml.template` file located in the root of validated patterns specifies the necessary secrets, detailing the required data types and keys for each pattern.
+
+In this template file, the `secrets` key contains an array of secrets. Each secret includes an array of `fields`, where each field has a `name` and `value` that describe the structure of the secret as used by the pattern.
+
+The Pattern framework will use the configured SecretStore backend to inject the secrets into the configured backend. Additionally, values can be retrieved in other ways than specifying a string value. This is convenient when the secret material likely already exists as a file elsewhere on the machine the pattern is being installed from, as SSH private and/or public keys already are. The framework also supports encoding the field using base64, which is required for binary data files, such as Red Hat Satellite-style manifests.
```
[the remark this answers is not in the document]
Comment (hbisht-RH-ccs) [reply modules/basic-secrets-configuration-in-a-clustergroup.adoc:11]: Done, thanks!

File: modules/basic-secrets-configuration-in-a-clustergroup.adoc [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+* *vaultPolicies*: (Optional): Allows injecting vault policies for automatic password generation; effective only when the secrets backend is vault. It’s a dictionary, allowing multiple policies.
+
+* *secrets*: The actual secrets to inject into the backend, with each secret having various sub-fields as described below.
+
+[id="sub-parameters-secret-file-structure"]
+== Sub-parameters
+
+* *name*: Indicates the name of the secret within the backend.
+
+* *fields*: Defines the fields within the secret object Each field has a name and a value. In the template file, the value is used as a placeholder to define the use of the field. This is used by the framework in error messages related to the field.
```
[the remark this answers is not in the document]
Comment (hbisht-RH-ccs) [reply modules/basic-secrets-configuration-in-a-clustergroup.adoc:46]: Missed that. Corrected! Thanks!

File: modules/basic-secrets-configuration-in-a-clustergroup.adoc [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+* *targetNamespaces*: (Optional, effective with “kubernetes” or “none” backends): Secret objects will be created in all specified namespaces. Multiple namespaces are allowed as the same secret material may need to be injected into multiple secret objects in different namespaces.
```
[the remark this answers is not in the document]
Comment (hbisht-RH-ccs) [reply modules/basic-secrets-configuration-in-a-clustergroup.adoc:48]: Done.

File: modules/letsencrypt-secrets-configuration-in-a-clustergroup.adoc [hunk 5]
```diff
@@ -0,0 +1,47 @@
+:_content-type: CONCEPT
+:imagesdir: ../../images
+
+[id="letsencrypt-secrets-configuration-in-a-clustergroup"]
+= Letsencrypt  secrets configuration in a ClusterGroup
```
[the remark this answers is not in the document]
Comment (hbisht-RH-ccs) [reply modules/letsencrypt-secrets-configuration-in-a-clustergroup.adoc:5]: Opps... corrected. Thanks!

File: modules/secret-store-configuration-in-a-clustergroup.adoc [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+----
+
+[id="secret-backends"]
+== Secrets backends in the Validated Patterns framework
+
+The validated patterns framework currently supports three secrets storage schemes:
+
+* *secrets-backend-vault*: This is the default scheme where the community edition of HashiCorp Vault and the External Secrets Operator are installed. Secrets are stored in vault and then projected into the pattern via External Secrets objects. This approach ensures versionable secrets without exposing them in a Git repository.
+
+* *secrets-backend-kubernetes*: The kubernetes secrets backend also uses the community External Secrets Operator, but rather than injecting secrets into Vault (which is not installed when this backend is selected), secrets are injected into a separate kubernetes namespace. ESO still brokers the projection of these secrets into the pattern.
```
[the remark this answers is not in the document]
Comment (hbisht-RH-ccs) [reply modules/secret-store-configuration-in-a-clustergroup.adoc:39]: Done, Thanks!

File: modules/secret-store-configuration-in-a-clustergroup.adoc [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+
+* *secrets-backend-none*: In this scheme, secret objects are directly injected into the pattern, based on the values-secret configuration. Neither Vault nor the External Secrets Operator are installed. This method is provided for users who require full Red Hat Supportability for their pattern installations.
+
+
+[id="switching-between-secrets-backends"]
+== Switching between secrets backends
+
+The framework provides Makefile-based mechanisms for changing secrets storage schemes. It is recommended to decide on a secrets storage mechanism before installing the pattern.  To configure your pattern with a different secrets storage scheme, run the appropriate make target. This is only necessary if you want to switch from the vault scheme, as all patterns come configured with the vault backend by default. The Makefile targets will modify the values file in your pattern repository, which you must then commit and push to take effect. The changes made by the Makefile targets will be displayed in "diff" format.
```
[the remark this answers is not in the document]
Comment (hbisht-RH-ccs) [reply modules/secret-store-configuration-in-a-clustergroup.adoc:47]: Done, thanks!

File: content/learn/secrets-management-in-the-Validated-Patterns-framework.adoc (hunk 1 above)
Comment (danmacpherson) [reply content/learn/secrets-management-in-the-Validated-Patterns-framework.adoc:4]: I had a think about this and I can see where @mhjacks is coming from. I do think when a customer comes to our site and is looking for information of secrets configuration, they'd be more likely to go to the Secrets section to find this content. Let's put it under Secrets (first sub menu item, above the Hashicorp Vault page) and we'll assess how to better structure our Learn content going forward.

Comment (hbisht-RH-ccs) [reply content/learn/secrets-management-in-the-Validated-Patterns-framework.adoc:4]: Thanks, @danmacpherson @mhjacks, I'll move it under secrets.

**Comment by danmacpherson** [comment]
Change requested from @mhjacks implemented with last commit from @hbisht-RH-ccs

**Pull request merged**